packages/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx

```diff
@@ -5,28 +5,53 @@ import styles from '@patternfly/react-styles/css/components/NotificationDrawer/n
 
 import { Text, TextVariants } from '../Text';
 
+interface NotificationDrawerCloseButtonProps extends React.HTMLProps<HTMLDivElement> {
+  /** Optional icon other than x */
+  icon?: React.ReactNode;
+  /**  Callback for when close button is clicked */
+  onClose: () => void;
+}
+
 export interface NotificationDrawerHeaderProps extends React.HTMLProps<HTMLDivElement> {
   /**  Content rendered inside the drawer */
   children?: React.ReactNode;
   /**  Additional classes for notification drawer header. */
   className?: string;
+  /** Optional icon other than x */
+  closeIcon?: React.ReactNode;
   /**  Notification drawer heading count */
   count?: number;
+  /**  Notification drawer heading custom text which can be used instead of providing count/unreadText */
+  customText?: string;
+  /**  Callback for when close button is clicked */
+  onClose?: () => void;
   /**  Notification drawer heading title */
   title?: string;
   /**  Notification drawer heading unread text used in combination with a count */
   unreadText?: string;
-  /**  Notification drawer heading custom text which can be used instead of providing count/unreadText */
-  customText?: string;
 }
 
+const NotificationDrawerCloseButton: React.FunctionComponent<NotificationDrawerCloseButtonProps> = ({
+  icon,
+  onClose
+}: NotificationDrawerCloseButtonProps) => (
+  <div>
+    <button className="pf-c-button pf-m-plain" type="button" aria-label="Close" onClick={onClose}>
```

Comment:
the aria-label needs to be customizable for localization.

Reply:
Added custom aria label prop for the button.